v1/plugins/plugins.go

```diff
@@ -604,6 +603,22 @@ func (m *Manager) InterQueryBuiltinCacheConfig() *cache.Config {
 	return m.interQueryBuiltinCacheConfig
 }
 
+// GetConfig returns the manager's configuration.
+// This is locked and is the supported way to access the manager's config.
+func (m *Manager) GetConfig() *config.Config {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+	return m.Config
+}
+
+// SetConfig sets the manager's configuration.
+// This is locked and is the supported way to set the manager's config.
+func (m *Manager) SetConfig(config *config.Config) {
```

Comment:
I have added this but it's only used in a test... so unsure if we should keep it.